core/components/atoms/tooltip/action.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Tooltip from './tooltip'
+
+class ActionTooltip extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { tooltipContent: null, asyncTaskIsRunnning: false }
+  }
+
+  /**
+   * If you provide a string as a content prop,
+   * this function will transform it to a
+   * `{ default: string }` shape.
+   */
+  preprocessContent() {
+    const rawContent = this.props.content
+    return typeof rawContent === 'string' ? { default: rawContent } : rawContent
+  }
+
+  /**
+   * Sets the tooltip inner text to `newContent`
+   * and resets the content to `props.content.default` after `resetDelay`.
+   * @param {string} newContent - Text to be set in the tooltip
+   * @param {string} resetDelay - Time in milliseconds to reset back to default text.
+   */
+  setTooltipContent(newContent, { resetDelay = 3000 } = {}) {
+    const content = this.preprocessContent()
+    this.setState({ tooltipContent: newContent })
+    setTimeout(() => this.setState({ tooltipContent: content.default }), resetDelay)
```

Comment:
I'd save this timeout id somewhere if possible, and on `componentWillUnmount` just reset everything?

Reply:
should we handle something on `onMouseLeave` or whatever trigger we have for the tooltip, cancel this timeout?